ring_buffer/ring_buffer.py

```diff
@@ -1,9 +1,67 @@
+# class RingBuffer:
+#     def __init__(self, capacity, data=[]):
+#         self.index = 0
+#         self.capacity = capacity
+#         self.data = list(data)[capacity:]
+
+#     def append(self, item):
+#         if len(self.data) == self.capacity:
+#             self.data[self.index] = item
+#         else:
+#             self.data.append(item)
+#         self.index = (self.index + 1) % self.capacity
+
+#     def get(self):
+#         return(self.data)
+
+class Node:
+    def __init__(self, value):
+        self.value = value
+        self.next = None
+
+    def get_value(self):
+        return self.value
+
+    def get_next(self):
+        return self.next
+
+    def set_next(self, new_next):
+        self.next = new_next
+
+
 class RingBuffer:
     def __init__(self, capacity):
-        pass
+        self.capacity = capacity
+        self.storage = []
+        self.index = 0
 
     def append(self, item):
-        pass
+        # Check if storage is empty
+        if len(self.storage) == 0:
```

Comment:
Could have achieved the same results by check if the return value of len(self.storage) is less than the capacity. If this condition is true then we know to append that item to the storage.